### PR TITLE
Fix missing null initialization of `pindexLastTip` pointer in ThreadStartWalletNotifier

### PR DESCRIPTION
**Pull request by nuttycom**

### CODE REVIEW

**Review by ebfull — APPROVED** [review]

File: src/init.cpp [hunk 1]
```diff
@@ -604,7 +604,7 @@ void CleanupBlockRevFiles()
 
 void ThreadStartWalletNotifier()
 {
-    CBlockIndex *pindexLastTip;
+    CBlockIndex *pindexLastTip{nullptr};
```
Comment (str4d) [inline src/init.cpp:607]: I love C++

**Review by str4d — APPROVED** [review]
utACK

File: src/init.cpp [hunk 2]
```diff
@@ -623,7 +623,7 @@ void ThreadStartWalletNotifier()
         }
 
         if (walletBestBlockHash.has_value()) {
-            int64_t slept;
+            int64_t slept{0};
```
Comment (str4d) [inline src/init.cpp:626]: I ❤️‍🔥 C++

**Review by daira — COMMENTED** [review]
I think it is not possible to get past the loop starting on line 649 without `pIndexLastTip` being initialized at line 656. (If the best block hash isn't found then the loop continues, and if a timeout occurs then we return from the function; the only way to exit the loop normally is by the break after `pIndexLastTip` is initialized.) If for the sake of argument we did, then the access in `chainActive.FindFork` (which always uses its argument) called at line 678 would be undefined behaviour. I suggest keeping the nullptr initialization but adding an assertion that it is non-null just after the loop.

**Review by str4d — APPROVED** [review]
utACK

**Pull request merged**